rwslib/builders/admindata.py

```diff
@@ -0,0 +1,395 @@
+        self._user_type = value
+
+    def build(self, builder):
+        """Build XML by appending to builder"""
+        params = dict(OID=self.oid)
+        if self.user_type:
+            params.update(dict(UserType=self.user_type.value))
+        builder.start(self.__class__.__name__, params)
+        # build the children
+        for child in ('login_name', 'display_name', 'full_name', 'first_name', 'last_name',
+                      'organisation'):
+            if getattr(self, child) is not None:
+                getattr(self, child).build(builder)
+        for address in self.addresses:
+            address.build(builder)
+        for email in self.emails:
+            email.build(builder)
+        for phone in self.phones:
+            phone.build(builder)
+        for location in self.locations:
+            location.build(builder)
+        builder.end(self.__class__.__name__)
+
+    def __lshift__(self, other):
+        """Override << operator"""
+        if not isinstance(other, (LoginName, DisplayName, FullName, FirstName, LastName, Organization,
+                                  Address, Email, Phone, LocationRef)):
+            raise ValueError('{0} cannot accept a {1} as a child element'.format(self.__class__.__name__,
+                                                                                 other.__class__.__name__))
+        self.set_list_attribute(other, Email, 'emails')
+        self.set_list_attribute(other, Address, 'addresses')
+        self.set_list_attribute(other, LocationRef, 'locations')
+        self.set_list_attribute(other, Phone, 'phones')
+        self.set_single_attribute(other, LoginName, 'login_name')
+        self.set_single_attribute(other, DisplayName, 'display_name')
+        self.set_single_attribute(other, FullName, 'full_name')
+        self.set_single_attribute(other, FirstName, 'first_name')
+        self.set_single_attribute(other, LastName, 'last_name')
+        self.set_single_attribute(other, Organization, 'organisation')
+        return other
+
+
+class SimpleChildElement(ODMElement):
+    """
+    Generic Element
+    """
+    def __init__(self, text):
+        self.text = text
+
+    def build(self, builder):
+        """
+        Build the element
+        :param builder: 
+        :return: 
+        """
+        builder.start(self.__class__.__name__)
+        builder.data(self.text)
+        builder.end(self.__class__.__name__)
+
+
+class LoginName(SimpleChildElement):
+    """
+    The user's login identification.
+    """
+    pass
+
+
+class DisplayName(SimpleChildElement):
+    """
+    A short displayable name for the user.
+    """
+    pass
+
+
+class FullName(SimpleChildElement):
+    """
+    The user's full formal name.
+    """
+    pass
+
+
+class FirstName(SimpleChildElement):
+    """
+    The user's initial given name or all given names.
+    """
+    pass
+
+
+class LastName(SimpleChildElement):
+    """
+    The user's surname (family name).
+    """
+    pass
```

Comment:
Small thing but in some of these you ```pass``` and in others you don't.

Reply:
Good point, will make it more consistent